src/IO/tests/lzma_buffers.cpp

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+#include <iomanip>
+
+#include <Common/Stopwatch.h>
+#include <IO/WriteBufferFromFile.h>
+#include <IO/ReadBufferFromFile.h>
+#include <IO/LzmaReadBuffer.h>
+#include <IO/LzmaWriteBuffer.h>
+#include <IO/WriteHelpers.h>
+#include <IO/ReadHelpers.h>
+
+int main(int, char **)
```

Comment:
Did you run it? (And what are the results in comparison with `xz`?)

Reply:
Yes, I ran. But I suspect, that my implementation is not completely optimal:
`(base) [centos@clickhouse build]$ ./src/IO/tests/lzma_buffers
Writing done. Elapsed: 12.78 s., 6.17 MB/s
Reading done. Elapsed: 0.79 s., 100.24 MB/s`

`(base) [centos@clickhouse build]$ ./src/IO/tests/zlib_buffers
Writing done. Elapsed: 8.04 s., 110.55 MB/s
Reading done. Elapsed: 4.44 s., 200.08 MB/s`

Reply:
It's hard to say... Better to compare with `xz` tool, and if performance should not be significantly lower.